src/fhirtypes/ElementDefinition.ts

```diff
@@ -1592,6 +1610,18 @@ export class ElementDefinition {
     if (isEmpty(validTypes)) {
       throw new CodedTypeNotFoundError(this.type ? this.type.map(t => t.code) : []);
     }
+    // Check if a CodeableReference is attempting to bind directly to the concept element
+    if (
+      this.type.length === 1 &&
+      this.type[0].code === 'CodeableConcept' &&
+      this.path.endsWith('.concept') &&
+      this.parent()?.type?.[0]?.code === 'CodeableReference'
+    ) {
+      logger.error(
+        'Binding constraints on CodeableReference elements should not be applied directly on the .concept element.',
```

Comment:
Similar to the other message, we need to also tell them what they _should_ do. If you liked my other suggestion, this one kind of mirrors it:
```suggestion
        'Applying value set bindings to a CodeableReference element's underlying .concept path is not allowed. Instead, apply the binding directly to the CodeableReference element.',
```